salt/config/__init__.py

```diff
@@ -3273,10 +3273,11 @@ def api_config(path):
     '''
     # Let's grab a copy of salt's master default opts
     defaults = DEFAULT_MASTER_OPTS
+    opts = client_config(path, defaults=defaults)
     # Let's override them with salt-api's required defaults
-    defaults.update(DEFAULT_API_OPTS)
+    opts.update(DEFAULT_API_OPTS)
```

Comment:
Applying the defaults against loaded opts is not good. You override configured settings with the defaults.